### DISCUSSION TITLE
What is the best approach to prevent iOS back swipe navigation and animation while still detecting it?

### DISCUSSION THREAD

**Discussion by wtfabio** [react-navigation/react-navigation#10309] [Q&A]
### Use case

I have a stack navigator from `@react-navigation/stack` version 6.x, containing `ScreenA` and `ScreenB`.
A button in `ScreenA` leads to `ScreenB`.

At `ScreenB`, I want to be able to detect the iOS back swipe gesture in order to perform a custom action, while:
1. preventing the default behavior (which is navigating back to the previous screen)
2. disabling the iOS swipe animation

What is the best approach to accomplish that? 

I can achieve what I want by disabling the gesture handler altogether in the stack screen options for `ScreenB` and then using `react-native-gesture-handler` to detect the "[fling](https://docs.swmansion.com/react-native-gesture-handler/docs/gesture-handlers/api/fling-gh)" gesture (back swipe). But I'm not sure if this is the best practice.

File: `StackScreens.tsx`
```tsx
<Stack.Navigator>
   <Stack.Screen name="ScreenA" component={ScreenA} />
   <Stack.Screen
     name="ScreenB"
     component={ScreenB}
     options={{ gestureEnabled: false }}
   />
</Stack.Navigator>
```
File: `ScreenB.tsx`
```tsx
<FlingGestureHandler
      direction={Directions.RIGHT}
      onHandlerStateChange={({ nativeEvent }): void => {
        // Custom action
      }}
    >
   <View> {/* View content */} </View>
</FlingGestureHandler>
```

P.S.: it's important to mention that using a navigation listener to listen to the `beforeRemove` event [as described in the docs](https://reactnavigation.org/docs/preventing-going-back/) was not sufficient to disable the iOS back swipe animation, although it did prevent the navigation.

Appreciate any help from the community!
Have a nice weekend :man_technologist:

**Comment by truongnguyen3** [comment]
I set `gestureEnabled` to false to disable  iOS back swipe navigation and animation. 
But what's the best practice to handle iOS back swipe navigation correctly?
